search_test.go

```diff
@@ -0,0 +1,1205 @@
+package redis_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/bsm/ginkgo/v2"
+	. "github.com/bsm/gomega"
+	"github.com/redis/go-redis/v9"
+)
+
+func WaitForIndexing(c *redis.Client, index string) {
+	for {
+		res, err := c.FTInfo(context.Background(), index).Result()
+		Expect(err).NotTo(HaveOccurred())
+		if res["indexing"].(float64) == 0 {
+			return
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+}
+
+var _ = Describe("RediSearch commands", Label("search"), func() {
+	ctx := context.TODO()
+	var client *redis.Client
+
+	BeforeEach(func() {
+		client = redis.NewClient(&redis.Options{Addr: ":6379"})
+		Expect(client.FlushDB(ctx).Err()).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		Expect(client.Close()).NotTo(HaveOccurred())
+	})
+
+	It("should FTCreate and FTSearch WithScores ", Label("search", "ftcreate", "ftsearch"), func() {
+		val, err := client.FTCreate(ctx, "txt", &redis.FTCreateOptions{}, &redis.SearchSchema{FieldName: "txt", FieldType: "TEXT"}).Result()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(val).To(BeEquivalentTo("OK"))
+		WaitForIndexing(client, "txt")
+		client.HSet(ctx, "doc1", "txt", "foo baz")
+		client.HSet(ctx, "doc2", "txt", "foo bar")
+		res, err := client.FTSearchWithArgs(ctx, "txt", "foo ~bar", &redis.FTSearchOptions{WithScores: true}).Result()
+		Expect(err).NotTo(HaveOccurred())
+		searchResult := res.(map[interface{}]interface{})
+		Expect(searchResult["total_results"]).To(BeEquivalentTo(int64(2)))
+		Expect(searchResult["results"].([]interface{})[0].(map[interface{}]interface{})["id"]).To(BeEquivalentTo("doc2"))
```

Comment:
This looks uncastable. can you try?

Reply:
IMHO points to the need for helpers on the return

Reply:
In Go, handling unstable return types like in the FTSearch function is challenging due to its static typing. Creating a helper function to determine the return type isn’t feasible due to the dynamic nature of the return type and the infinite possible variations.